### PR TITLE
Add a collect.binlog_size flag that enables an aggregated binlog size gauge

### PR DESCRIPTION
**Pull request by rtreffer**
SHOW MASTER LOGS or SHOW BINARY LOGS can be used to retrieve a
list of all binary logs (including size). Summing up the sizes
gives a rough counter for the current binlog size.

See https://dev.mysql.com/doc/refman/5.7/en/show-binary-logs.html
for details.

This gauge simply sums the File_size column. (Note that the total size will grow/shrink as new binlogs are written and old ones are purged)

Required mysql permission: SUPER or REPLICATION CLIENT

### CODE REVIEW

**Comment by SuperQ** [comment]
Nice.  Maybe also count the number of binlog files?

**Comment by rtreffer** [comment]
@SuperQ good idea, added.

**Comment by rtreffer** [comment]
Is there anything left that I should do for this pr?

File: mysqld_exporter.go [hunk 1]
```diff
@@ -154,6 +160,16 @@ var landingPage = []byte(`<html>
 
 // Metric descriptors for dynamically created metrics.
 var (
+	binlogSizeDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, binlog, "size"),
```
Comment (SuperQ) [inline mysqld_exporter.go]: this should be `size_total`

File: mysqld_exporter.go [hunk 2]
```diff
@@ -154,6 +160,16 @@ var landingPage = []byte(`<html>
...
+		"Combined size of all registered binlog files.",
+		[]string{}, nil,
+	)
+	binlogFilesDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, binlog, "files"),
```
Comment (SuperQ) [inline mysqld_exporter.go]: Same here, `files_total`

File: mysqld_exporter.go [hunk 3]
```diff
@@ -78,6 +82,7 @@ const (
 	informationSchema = "info_schema"
 	performanceSchema = "perf_schema"
 	slaveStatus       = "slave_status"
+	binlog            = "binlog"
```
Comment (SuperQ) [inline mysqld_exporter.go]: I'm trying to decide if I like this name.  What do you think of calling this `master_logs` to match the source of the data.  This way the final metric name would be `mysql_master_logs_size_total`.

**Comment by SuperQ** [comment]
Everything is good code wise. I had a think about the naming, some suggestions added.  Let me know what you think.

File: mysqld_exporter.go (hunk 1 above)
Comment (SuperQ) [inline mysqld_exporter.go]: @juliusv Oh wait, is this the correct way to name gauge metrics?

**Comment by rtreffer** [comment]
I have no preference on the names.

The files are usually called "mysql-bin" on disk, the options are called binlog (e.g. max_binlog_size) and the command is either "show master logs" or "show binary logs".

How about changing the query to SHOW BINARY LOGS ?

File: mysqld_exporter.go (hunk 1 above)
Comment (matthiasr) [inline mysqld_exporter.go]: No, `_total` is for counters ("total over all time ")

File: mysqld_exporter.go [hunk 4]
```diff
@@ -154,6 +160,16 @@ var landingPage = []byte(`<html>
 
 // Metric descriptors for dynamically created metrics.
 var (
+	binlogSizeDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, binlog, "size_total"),
```
Comment (SuperQ) [inline mysqld_exporter.go]: Oops, I guess I was wrong about `_total` for gauges.  Sorry.

**Comment by SuperQ** [comment]
Ahh, I see, yes, it makes sense to use `binlog` if many of the control flags are named this way.  `SHOW BINARY LOGS` would be fine, is more consistent with the internal naming.

**Comment by SuperQ** [comment]
Please squash the commits.

**Comment by rtreffer** [comment]
@SuperQ ok, will squash

What's the correct name for the gauge values? Was size and files correct? Or size + files_total? Or any other mix?

**Comment by SuperQ** [comment]
The previous names of `size` and `files` were correct.

**Comment by rtreffer** [comment]
Squashed.

**Comment by SuperQ** [comment]
:+1: :shipit:

File: mysqld_exporter.go (hunk 1 above)
Comment (juliusv) [inline mysqld_exporter.go]: But if this is a size gauge in bytes, add a unit suffix like `..._binlog_size_bytes`. If it's just a current count of something (like the `..._binlog_files` metric below, no suffix is needed).

**Pull request merged**